### PR TITLE
[IPU] Do not use `DistributedSampler`

### LINKED ISSUE #10694: Incorrectly applying DistributedSampler with poptorch.DataLoader
## 🐛 Bug

There are two-interrelated issues when implementing a `LightningDataModule` that directly creates an instance of `poptorch.DataLoader` .  This issue tracks the following error:

```
pytorch_lightning.utilities.exceptions.MisconfigurationException: Trying to inject `DistributedSampler` into the `DataLoader` instance. This would fail as some of the `__init__` arguments are not available as instance attributes. The missing attributes are ['options']. HINT: If you wrote the `DataLoader` class, define `self.missing_arg_name` or manually add the `DistributedSampler` as: `DataLoader(dataset, sampler=DistributedSampler(dataset)
```

It is incorrect to use the `DistributedSampler` with a `poptorch.DataLoader` as this sampling strategy does not apply.   

### To Reproduce

repro.py
```python
import torch
import poptorch
from torch.utils.data import Dataset

from pytorch_lightning import LightningModule, LightningDataModule
from pytorch_lightning.utilities.cli import LightningCLI


class RandomDataset(Dataset):
    def __init__(self, size, length):
        self.len = length
        self.data = torch.randn(length, size)

    def __getitem__(self, index):
        return self.data[index]

    def __len__(self):
        return self.len


class BoringDataModule(LightningDataModule):
    def get_options(self, training=False):
        plugin = self.trainer.accelerator.training_type_plugin

        if training:
            return plugin.training_opts

        return plugin.inference_opts

    def train_dataloader(self):
        opts = self.get_options(training=True)
        return poptorch.DataLoader(options=opts,
                                   dataset=RandomDataset(32, 64),
                                   batch_size=2)

    def val_dataloader(self):
        opts = self.get_options(training=False)
        return poptorch.DataLoader(options=opts,
                                   dataset=RandomDataset(32, 64),
                                   batch_size=2)

    def test_dataloader(self):
        opts = self.get_options(training=False)
        return poptorch.DataLoader(options=opts,
                                   dataset=RandomDataset(32, 64),
                                   batch_size=2)


class BoringModel(LightningModule):
    def __init__(self):
        super().__init__()
        self.layer = torch.nn.Linear(32, 2)

    def forward(self, x):
        return self.layer(x)

    def training_step(self, batch, batch_idx):
        loss = self(batch).sum()
        self.log("train_loss", loss)
        return poptorch.identity_loss(loss, reduction="none")

    def validation_step(self, batch, batch_idx):
        loss = self(batch).sum()
        self.log("valid_loss", loss)
        return loss

    def test_step(self, batch, batch_idx):
        loss = self(batch).sum()
        self.log("test_loss", loss)

    def configure_optimizers(self):
        return torch.optim.SGD(self.layer.parameters(), lr=0.1)


cli = LightningCLI(model_class=BoringModel, datamodule_class=BoringDataModule)

```

repro.yml
```
trainer:
  limit_train_batches: 1.0
  limit_val_batches: 1.0
  num_sanity_val_steps: 0
  max_epochs: 1
  enable_model_summary: false
  ipus: 1
  strategy:
    class_path: pytorch_lightning.plugins.IPUPlugin
    init_args:
      device_iterations: 8
```

```bash
python repro.py fit --config repro.config 
```

### Expected behavior

* Directly instantiating a `poptorch.DataLoader` is supported from within a `LightningDataModule`
* `DistributedSampler` is never used with `poptorch.DataLoader`  

### Environment

- PyTorch Lightning Version: 1.5.2
- PyTorch Version: 1.10.0
- Python version (e.g., 3.9): 3.6.8
- OS (e.g., Linux): ubuntu 18
- CUDA/cuDNN version:
- GPU models and configuration:
- How you installed PyTorch (`conda`, `pip`, source): pip
- If compiling from source, the output of `torch.__config__.show()`:
- Any other relevant information:

cc @SeanNaren

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Fixes #10694

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [n/a] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

cc @borda @justusschock @awaelchli @ninginthecloud

### CODE REVIEW

File: pytorch_lightning/strategies/ipu.py [hunk 1]
```diff
@@ -214,10 +214,11 @@ def lightning_module(self) -> Optional["pl.LightningModule"]:
     def _convert_to_poptorch_loader(
         self, dataloader: DataLoader, sampler, mode: Optional[RunningStage] = None
     ) -> "poptorch.DataLoader":
-        dl_kwargs = _get_dataloader_init_kwargs(dataloader, sampler)
-        # Override to drop last uneven batch, as IPUs does not support uneven inputs.
-        dl_kwargs["drop_last"] = True
```
Comment (carmocca) [inline pytorch_lightning/strategies/ipu.py:219]: I'm unsure why we were dropping this key, they do support it:

https://docs.graphcore.ai/projects/poptorch-user-guide/en/latest/reference.html#poptorch.DataLoader

And drop_last does not mean uneven inputs, but whether we drop the uneven ones or pad with repeated samples

File: pytorch_lightning/strategies/ipu.py [hunk 2]
```diff
@@ -214,10 +214,11 @@ def lightning_module(self) -> Optional["pl.LightningModule"]:
...
+        if isinstance(dataloader, poptorch.DataLoader):
+            # the user is returning the `poptorch.DataLoader` directly, don't change anything.
+            return dataloader
```
Comment (carmocca) [inline pytorch_lightning/strategies/ipu.py:219]: This is not strictly related to the bugfix, but we probably should not recreate the dataloader if the user already created a `poptorch.DataLoader`

**Review by kaushikb11 — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]

**Review by SeanNaren — APPROVED** [review]
